packages/gatsby/src/services/build-schema.ts

```diff
@@ -0,0 +1,19 @@
+import { IBuildContext } from "./"
+
+import { build } from "../schema"
+import reporter from "gatsby-cli/lib/reporter"
+
+export async function buildSchema({
+  store,
+  parentSpan,
+}: Partial<IBuildContext>): Promise<void> {
+  if (!store) {
```

Comment:
Wouldn't TS Catch this error? I'm fine leaving the error in, just wondering.

Reply:
The context has the store as nullable, because it's not there when the state machine starts. We add it early on, but because the context is always the same type, we need to add an explicit check. Having the runtime check also helps to debug state machine issues